### PR TITLE
Fix incorrect comparision between symbols and strings in deadlines

### PR DESCRIPTION
**Pull request by byucesoy**
In register_deadline function, we have a logic where we don't allow extending the deadline if the target is same. However, we were comparing symbols with strings to decide if the target is same, which always results in us deciding the targets are different and extending the deadline. This commit fixes it by using strings for deadline targets.

### CODE REVIEW

File: migrate/20241204_vm_add_vcpus_memory.rb [hunk 1]
```diff
@@ -2,9 +2,5 @@
 
 Sequel.migration do
   change do
-    alter_table(:vm) do
```
Comment (enescakir) [inline migrate/20241204_vm_add_vcpus_memory.rb:5]: Why did you change migrations

Comment (byucesoy) [reply migrate/20241204_vm_add_vcpus_memory.rb:5]: Migrations failed on my setup because they are applied in the wrong order. I made this change to fix them in my environment, but didn't intend to commit the changes. I reverted it back now.

**Comment by enescakir** [comment]
@jeremyevans has a whole chapter in his book about symbols and strings. He might have some reviews, but I believe we should use strings whenever possible, especially when saving to the database. The database always returns strings, and using symbols complicates things

**Review by jeremyevans — COMMENTED** [review]
As @enescakir mentioned, you should use strings if these values are stored in the database.  This also applies to other external storage cases, such as the session or flash.

**Comment by byucesoy** [comment]
Yep, they are used in the database, so this PR starts to use strings instead of symbols.

**Review by enescakir — APPROVED** [review]
